src/ds/prob/intro.md

```diff
@@ -0,0 +1,39 @@
+Redis supports **probabilistic data structures** that provide reasonable yet approximate results quickly and with significantly less memory overhead than their deterministic counterparts.
+The following data structures trade perfect accuracy for extreme memory efficiency, making them especially useful for big data and streaming applications:
+
+* Bloom filter
+* Cuckoo filter
+* Count-Min sketch
+* Top-k
+
+**Bloom filter:**
+* Enables you to check if an element is present in a set using a fixed size (sublinear) memory space, sacrificing some precision.
+* Uses multiple hash functions to create a representation of the element in a bit array.
+* Answers the question "Is this element present in the set?".
+* A negative answer means that the element is definitively not present in a set.
+* A positive answer means that the element is most likely present in the set. Note: false positives are possible.
+* The error rate can be configured when the filter is created.
+
+**Cuckoo filter:**
+* Fulfills the same mission as Bloom filters, but also allow for item deletion.
+* Provides better results than Bloom filters under certain circumstances.
+
+**Count-min sketch**
+* Estimates the frequency of events/elements in a stream of data.
+* Consumes a stream of events/elements and keeps estimates of their frequency.
+* Can only be trusted for counts above a certain threshold.
+* Biased: results can be higher than reality, but never lower.
+
+**Top k**
+* Used to estimate the **k** highest-rank elements in a stream.
+* Optimized for large flows; biased on small flows.
+* Provide notification when an element is removed from the top k list.
```

Comment:
Top-K
I don't think that 'notifications' is the correct term here. It is not keyspace notifications, but part of the command's reply